Sort series in the same order as legend in graph panel

bergquist opened the pull request:
closes #9538

bergquist committed:
graph: sort series in the same order as legend

closes #9538

torkelo commented on public/app/plugins/panel/graph/legend.js:
```diff
@@ -80,13 +80,13 @@ function (angular, _, $) {
           panel.legend.sortDesc = !panel.legend.sortDesc;
           panel.legend.sort = stat;
-          render();
+          ctrl.events.emit('render');
```
this should be ctrl.render()   (it does the emit)

bergquist commented:
ready for review!

bergquist committed:
style: use function arrows

torkelo commented on public/app/plugins/panel/graph/graph.ts:
```diff
@@ -343,7 +343,8 @@ function graphDirective($rootScope, timeSrv, popoverSrv, contextSrv) {
 
-        sortedSeries = _.sortBy(data, function(series) { return series.zindex; });
+        sortedSeries = _.sortBy(data, (series) => { return series.zindex; });
+        sortSeries(sortedSeries, data, ctrl.panel);
```
Looks like it sort twice here (when legend sort is enabled), maybe move zindex sort into sortSeries?

bergquist replied:
Should series override (zindex) or sort as legend rank highest?

I think we should execute zindex sorting at the end.

torkelo replied:
Yes. That might be best. Do it inside sortSeries after any legend sort.

torkelo commented on public/app/plugins/panel/graph/tab_display.html:
```diff
@@ -71,12 +71,14 @@ <h5 class="section-heading">Hover tooltip</h5>
 		<div class="section gf-form-group">
 			<h5 class="section-heading">Stacking & Null value</h5>
-			<gf-form-switch class="gf-form" label="Stack" label-class="width-7" checked="ctrl.panel.stack" on-change="ctrl.render()">
+			<gf-form-switch class="gf-form" label="Sort as Legend" label-class="width-8" checked="ctrl.panel.sortAsLegend" on-change="ctrl.render()">
```
Think a dropdown open named "Stack Order" with two options "By Query" , "By Legend Order". But not sure if that is clearer. Also worth discussing, do we need an option for this? When stacking is this option 95% of the time always desired?

bergquist replied:
Good point!

I bet there might be cases where this is not the case but then people could use series override to get their series to behave exactly as they want.

bergquist committed:
graph: remove sort by legend option

sort series based on legend by default is sort order
is set and stack is enabled

bergquist committed:
graph: remove useless sort

bergquist committed:
graph: make code simpler

torkelo commented:
This is looking solid!
